src/main/java/frc/robot/classes/DebugLog.java

```diff
@@ -0,0 +1,54 @@
+package frc.robot.classes;
+
+import java.util.function.Consumer;
+
+import edu.wpi.first.networktables.GenericEntry;
+import edu.wpi.first.util.datalog.DataLog;
+import edu.wpi.first.wpilibj.DataLogManager;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Robot;
+
+public class DebugLog<T> {
+    private static DataLog datalog = DataLogManager.getLog();
+
+    private ShuffleboardTab networkTab;
+    private GenericEntry networkEntry;
+    private Integer localEntry;
+    private Consumer<T> localConsumer;
+
+    public DebugLog(T defaultValue, String name, SubsystemBase subsystem){
+        if(defaultValue instanceof Double){
+            localEntry = datalog.start("/" + subsystem.getName() + "/" + name, "double");
+            localConsumer = (a) -> datalog.appendDouble(localEntry, (Double)a, 0);
+        }
+        if(defaultValue instanceof String){
+            localEntry = datalog.start("/" + subsystem.getName() + "/" + name, "string");
+            localConsumer = (a) -> datalog.appendString(localEntry, (String)a, 0);
+        }
+        if(defaultValue instanceof Boolean){
+            localEntry = datalog.start("/" + subsystem.getName() + "/" + name, "boolean");
+            localConsumer = (a) -> datalog.appendBoolean(localEntry, (Boolean)a, 0);
+        }
+        if (localEntry != null) {
+            if(!Robot.isCompetition){
+                networkTab = Shuffleboard.getTab(subsystem.getName());
+                networkEntry = networkTab.add(name, defaultValue).getEntry();
```

Comment:
Since we encapsulate the GenericEntry, we can't add decorators anymore. Is there a way to support those?

Reply:
Any entry that uses decorators would be for the competition dashboard, right?
If so, that's not supported by this class, which is only for debug logs

Reply:
It reduces some flexibility, but its a reasonable limitation